arlo-client/src/components/Audit/useBallotManifest.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import { toast } from 'react-toastify'
+import { api } from '../utilities'
+import { IFileInfo } from './useJurisdictions'
+
+const loadBallotManifest = async (
+  electionId: string,
+  jurisdictionId: string
+): Promise<IFileInfo | null> => {
+  try {
+    return await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/ballot-manifest`
+    )
+  } catch (err) {
+    toast.error(err.message)
+    return null
+  }
+}
+
+const putBallotManifestFile = async (
+  electionId: string,
+  jurisdictionId: string,
+  csv: File
+): Promise<boolean> => {
+  const formData: FormData = new FormData()
+  formData.append('manifest', csv, csv.name)
+  try {
+    await api(
+      `/election/${electionId}/jurisdiction/${jurisdictionId}/ballot-manifest`,
+      {
+        method: 'PUT',
+        body: formData,
+      }
+    )
+    return true
+  } catch (err) {
+    toast.error(err.message)
+    return false
+  }
+}
+
+const useBallotManifest = (
+  electionId: string,
+  jurisdictionId: string
+): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
+  const [ballotManifest, setBallotManifest] = useState<IFileInfo | null>(null)
+
+  const uploadBallotManifest = async (csv: File): Promise<boolean> => {
+    // TODO poll for result of upload
```

Comment:
I think there will be an elegant and straightforward way to put the polling into this hook. It's not a fully baked idea yet but my intuition is feeling good about it. @MorganLove would love to run it by you

Reply:
@MorganLove Kind of like what I did here might work: https://github.com/votingworks/arlo/blob/master/arlo-client/src/components/Audit/useSetupMenuItems/index.ts

I definitely think it can be integrated here in an elegant fashion.